anx/provider/loadbalancer/address/address.go

```diff
@@ -196,7 +201,7 @@ func (m *mgr) allocateAddress(ctx context.Context, svc *v1.Service, fam v1.IPFam
 	// for every prefix, try to allocate an address from it, returning the first that works
 	for _, p := range prefixes {
 		if p.family == fam {
-			ip, err := p.allocateAddress(ctx, fam)
+			ip, err := p.allocateAddress(ctx, fam, m.api, m.ipam, m.vipAutoDiscover)
```

Comment:
instead of passing them there, I'd like `prefix` to be changed to have an array for usable addresses, filled by `m.prefixes()` which already has all info to fill that array

Reply:
having a flag per `address` if free or in-use could be good

Reply:
I think this is both a little premature and out of scope for this PR, we're not moving away from the 'one external address per cluster' limitation.
Also, prefixes could be dropped altogether once all clusters are migrated to tagged addresses?

Reply:
After our discussion I've introduced `prefix.addresses` for the list of potential external IP addresses in the prefix.